Holovibes/sources/data_struct/batch_input_queue.cc

```diff
@@ -121,70 +121,89 @@ void BatchInputQueue::stop_producer()
     }
 }
 
-void BatchInputQueue::enqueue(const void* const input_frame, const cudaMemcpyKind memcpy_kind)
+void BatchInputQueue::enqueue(const void* const frames, const cudaMemcpyKind memcpy_kind, const int nb_frame)
 {
     if ((memcpy_kind == cudaMemcpyDeviceToDevice || memcpy_kind == cudaMemcpyHostToDevice) && (device_ == Device::CPU))
         throw std::runtime_error("Input queue : can't cudaMemcpy to device with the queue on cpu");
 
     if ((memcpy_kind == cudaMemcpyDeviceToHost || memcpy_kind == cudaMemcpyHostToHost) && (device_ == Device::GPU))
         throw std::runtime_error("Input queue : can't cudaMemcpy to host with the queue on gpu");
 
-    if (curr_batch_counter_ == 0) // Enqueue in a new batch
-    {
-        // The producer might be descheduled before locking.
-        // Consumer might modified curr_batch_counter_ here by the resize
-        // However, it sets its value to 0 so the condition is still true.
-        m_producer_busy_.lock();
-        batch_mutexes_[end_index_].lock();
-    }
+    int frames_left = nb_frame;
 
-    // Critical section between enqueue (producer) & resize (consumer)
+    while (frames_left > 0)
+    {
+        if (curr_batch_counter_ == 0) // Enqueue in a new batch
+        {
+            // The producer might be descheduled before locking.
+            // Consumer might modified curr_batch_counter_ here by the resize
+            // However, it sets its value to 0 so the condition is still true.
+            m_producer_busy_.lock();
+            batch_mutexes_[end_index_].lock();
+        }
+        // Critical section between enqueue (producer) & resize (consumer)
 
-    // Static_cast to avoid overflow
-    char* const new_frame_adress =
-        data_.get() + ((static_cast<size_t>(end_index_) * batch_size_ + curr_batch_counter_) * fd_.get_frame_size());
+        uint frames_to_enqueue = std::min((uint)frames_left, batch_size_ - curr_batch_counter_);
```

Comment:
Casting directly like in C is not clean in C++. Use static_cast instead

Reply:
Done